queue/queue.py

```diff
@@ -1,15 +1,25 @@
 class Queue:
-  def __init__(self):
-    self.size = 0
-    # what data structure should we
-    # use to store queue elements?
-    self.storage = 
-
-  def enqueue(self, item):
-    pass
-  
-  def dequeue(self):
-    pass
-
-  def len(self):
-    pass
+    def __init__(self):
+        self.size = 0
+        # what data structure should we
+        # use to store queue elements?
+        self.storage = []
+        # super naive first pass implementation!
+
+    def enqueue(self, item):
+        self.storage.append(item)
+        return
+
+    def dequeue(self):
+        if len(self.storage) > 0:
+            return self.storage.pop(0)
```

Comment:
To elaborate on the above critique:

Since we are using array/list, popping an item off the front forces all other items to be re-index'd (a linear time operation).  If we were using a LinkedList, this would be a constant time operation instead.  This is one strong reason that we use a LinkedList as the data structure of choice for implementing a Queue or Stack.